### PR TITLE
[refactor]: Internalize loader api

### PR DESCRIPTION
**Pull request by varshaprasad96**
This PR intends to move the loader api to
internal. Only the necessary methods which
are needed for the api have been put into
`pkg/loader.go`.

### CODE REVIEW

**Comment by natasha41575** [comment]
Thanks for the PR @varshaprasad96! @koba1t could you give this a review? The related issue is https://github.com/kubernetes-sigs/kustomize/issues/3942

/assign @koba1t

**Review by koba1t — CHANGES_REQUESTED** [review]
@varshaprasad96 
I've commented on a minor change.
Could you check it?

**Comment by varshaprasad96** [comment]
@natasha41575 @annasong20 Could we merge this? Thanks!

**Comment by natasha41575** [comment]
@varshaprasad96 some tests are failing, could you address those?

File: api/pkg/loader/loader.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2019 The Kubernetes Authors.
```
Comment (natasha41575) [inline api/pkg/loader/loader.go:1]: nit: 2023

**Review by natasha41575 — COMMENTED** [review]
/lgtm
/approve

/hold

Feel free to remove the hold when you update my nit above and ping me for another lgtm.

File: api/pkg/loader/loader.go (hunk 1 above)
Comment (varshaprasad96) [reply api/pkg/loader/loader.go]: Addressed

**Pull request merged**